gdx/src/com/badlogic/gdx/graphics/g3d/utils/shapebuilders/RenderableShapeBuilder.java

```diff
@@ -136,71 +137,73 @@ public static void buildNormals (MeshPartBuilder builder, Renderable renderable,
 		mesh.getVertices(verticesOffset * attributesSize, verticesQuantity * attributesSize, vertices, 0);
 
 		for (int i = verticesOffset; i < verticesQuantity; i++) {
 			int id = i * attributesSize;
 
 			// Vertex position
-			tmpV0.set(vertices[id + positionOffset], vertices[id + positionOffset + 1], vertices[id + positionOffset + 2]);
+			data.tmpV0.set(vertices[id + positionOffset], vertices[id + positionOffset + 1], vertices[id + positionOffset + 2]);
 
 			// Vertex normal, tangent, binormal
 			if (normalOffset != -1) {
-				tmpV1.set(vertices[id + normalOffset], vertices[id + normalOffset + 1], vertices[id + normalOffset + 2]);
-				tmpV2.set(tmpV0).add(tmpV1.scl(vectorSize));
+				data.tmpV1.set(vertices[id + normalOffset], vertices[id + normalOffset + 1], vertices[id + normalOffset + 2]);
+				data.tmpV2.set(data.tmpV0).add(data.tmpV1.scl(vectorSize));
 			}
 
 			if (tangentOffset != -1) {
-				tmpV3.set(vertices[id + tangentOffset], vertices[id + tangentOffset + 1], vertices[id + tangentOffset + 2]);
-				tmpV4.set(tmpV0).add(tmpV3.scl(vectorSize));
+				data.tmpV3.set(vertices[id + tangentOffset], vertices[id + tangentOffset + 1], vertices[id + tangentOffset + 2]);
+				data.tmpV4.set(data.tmpV0).add(data.tmpV3.scl(vectorSize));
 			}
 
 			if (binormalOffset != -1) {
-				tmpV5.set(vertices[id + binormalOffset], vertices[id + binormalOffset + 1], vertices[id + binormalOffset + 2]);
-				tmpV6.set(tmpV0).add(tmpV5.scl(vectorSize));
+				data.tmpV5.set(vertices[id + binormalOffset], vertices[id + binormalOffset + 1], vertices[id + binormalOffset + 2]);
+				data.tmpV6.set(data.tmpV0).add(data.tmpV5.scl(vectorSize));
 			}
 
 			// World transform
-			tmpV0.mul(renderable.worldTransform);
-			tmpV2.mul(renderable.worldTransform);
-			tmpV4.mul(renderable.worldTransform);
-			tmpV6.mul(renderable.worldTransform);
+			data.tmpV0.mul(renderable.worldTransform);
+			data.tmpV2.mul(renderable.worldTransform);
+			data.tmpV4.mul(renderable.worldTransform);
+			data.tmpV6.mul(renderable.worldTransform);
 
 			// Draws normal, tangent, binormal
 			if (normalOffset != -1) {
 				builder.setColor(normalColor);
-				builder.line(tmpV0, tmpV2);
+				builder.line(data.tmpV0, data.tmpV2);
 			}
 
 			if (tangentOffset != -1) {
 				builder.setColor(tangentColor);
-				builder.line(tmpV0, tmpV4);
+				builder.line(data.tmpV0, data.tmpV4);
 			}
 
 			if (binormalOffset != -1) {
 				builder.setColor(binormalColor);
-				builder.line(tmpV0, tmpV6);
+				builder.line(data.tmpV0, data.tmpV6);
 			}
 		}
 	}
 
-	private static void ensureVerticesCapacity (int capacity) {
+	private static float[] ensureVerticesCapacity (int capacity, float[] vertices) {
 		if (vertices == null || vertices.length < capacity) vertices = new float[capacity];
+		return vertices;
 	}
 
-	private static void ensureIndicesCapacity (int capacity) {
+	private static short[] ensureIndicesCapacity (int capacity, short[] indices) {
```

Comment:
Maybe I'm wrong, but isn't this function unused?

Reply:
we did not alter the api, but @dar-dev can look into it

Reply:
it is. Let me review it

Reply:
same. No longer used.